### ISSUE TITLE
Host admin can still access financial data for previously hosted accounts

### ISSUE THREAD

**Issue by alanna** [opencollective/opencollective#5137]
## User story 
As a host admin, I need to be able to access full details of a Collective's transactions and expenses, even if it has been archived, so that I can gather historical information for things like tax reporting.

## Best solution for this problem
To the public, archived Collectives stay as they are now, but to the host admin, they display all the data available for active Collectives, with some visual indication that the Collective is in fact archived and all the action buttons inactive.

  P2 low frequency, high impact

**Comment by Betree** [comment]
I think we already display historical financial data for archived collectives? Like this one for example https://staging.opencollective.com/atomex#category-BUDGET the budget section is there. The transactions page is also accessible.

The only thing is that the transactions/expenses page don't show clearly that the collective is archived.

**Comment by alanna** [comment]
I was trying to look into past expense for https://opencollective.com/justice-reskill and I can't access anything.

<img width="755" alt="Screen Shot 2022-01-25 at 9 47 54 PM" src="https://user-images.githubusercontent.com/5498878/150943024-09ca32b1-696e-4751-8af3-0c3535d64654.png">

**Comment by Betree** [comment]
Thanks for the link, there's indeed a specificity with the expenses page that we need to address

**Comment by alanna** [comment]
@Betree this still isn't working for me but perhaps it just hasn't been deployed yet?

**Comment by Betree** [comment]
The bug indeed seems to be still there for https://opencollective.com/justice-reskill. I'll need to dig further

**Comment by Betree** [comment]
So the fix from https://github.com/opencollective/opencollective-frontend/pull/7382 did solve most cases.

The reason why it did not work for `justice-reskill` is that they've hidden their budget (only funds and projects are allowed to hide their budget/transactions history). Now of course, as a fiscal host admin, you should be allowed to see these past transactions and expenses when it's an account you've hosted in the past.

It's not a simple fix, as the place we're looking for these permissions in the frontend doesn't have this kind of history. I can suggest adding something along those lines to a new `Account.permissions` field that we'd use in the frontend:

<details>

<summary>Specs details</summary>

```es6
const Account = {
  ...
  permissions: {
    seeFinancialData: {
      async resolve(account) {
        if (hasBudgetSection(account)) {
          return true;
        } else {
          return isAdmin(account) || isHostAdmin(account) || await isAdminOfPreviousHost(account);
        }
      }
    }
  }
}
```

The `isAdminOfPreviousHost` part is the tricky one, as we have no quick way to see that. Maybe the best option is to list the unique `HostCollectiveId` from the transactions table?

</details>

@alanna If a hot fix is needed, I can propose to always allow access to financial data for root users.

**Comment by alanna** [comment]
Yes please do the hot fix. There is a dispute involving this collective and
we need access to historical data.

**Comment by Betree** [comment]
@alanna The hot fix is live

**Comment by jfdh** [comment]
**Here is another User Story related:**

As a host Admin, I need to access all the details of past paid expenses (including the attachment). Even if expenses are part of an _archived project_. This is important as I have to report to my accountants the address and the tax details of the payee, which are often only available on the attachment. 

Example with [this expense](https://opencollective.com/speakeasy-brussels/projects/nonsense-labyrinth/expenses/60754) where I can't access the attachment

**Comment by alanna** [comment]
Another related situation is that hosts need to be able to access full records for Collectives that were previously with that Host but have since moved to a new Host. I need to be able to access everything from when the Collective was under my Host.

**Comment by alanna** [comment]
@Betree we need to give access to this archived Collective's expenses to people with "accountant" team role. Can this please be enabled? We need to let them review transactions pretty urgently for this specific case.

**Comment by Betree** [comment]
The fix for root users was an easy one because we don't need to check permissions against the collective, we just allowed them to access the data of **any** archived collectives.
Now solving that for accountants basically means solving this issue: like for admins, we have to check the permissions against this specific collective. The root problem is that we remove the host when archiving, so it's not straightforward to check this permission. More info in https://github.com/opencollective/opencollective/issues/5137#issuecomment-1023058302.

@alanna Is there another way we could solve this? How many collectives are we talking about? Which parts do the accountants need access to (view transactions? download PDFs?)?

**Comment by alanna** [comment]
The real issue is a pretty deep one - at some level transactions/expenses
belong to the host as much as the collective, as in if a transaction goes
through while a collective is with a given host, that transaction always
needs to maintain association with that host because it went through their
bank account, even if the collective moves hosts later, or is archived.

It's for reasons like we have now - there's an issue and we have to look at
the transaction record historically. Our lawyer and accountant need to be
able to review everything, including transactions, expenses, and expense
attachments.

We absolutely need this for tax reporting as well. When OCF is audited for
2021, how will we provide access to receipts for expenses of collectives
who have been archived or moved hosts in the meantime? We are legally
obligated to maintain these records as a host, and our auditors must be
given access. The way they work is through randomly selecting transactions
and then digging up every detail about them, including stuff like Updates
associated with the Collective the expense is for, etc etc. If they can't
do this through the platform it's going to be painful.

For the urgent issue of this particular investigation, I have created a
data export from metabase for the lawyer, and I should be able to dig out
related attachments, etc, if needed. But she wants to be able to review
when the expense was submitted, by whom, when it was approved, and by whom,
and when and how it was paid. This level of detail is hard to achieve with
metabase, and of course is not accessible to third party hosts, so it's not
really a solution.

**Comment by Betree** [comment]
The specs proposed in https://github.com/opencollective/opencollective/issues/5137#issuecomment-1023058302 can be a good fix. It's not perfect, as it will basically allow host admins/accountants to access **all** financial activity of collectives they have hosted in the past (including what happened with the new host), whereas we would want something more like accessing only the activity for the period when they were hosted.

I can try to prioritize in the next few weeks if it's time sensistive.

**Comment by alanna** [comment]
If the lawyer is OK with the data export, then this is less urgent. I'll report back soon. Long term we do need to solve this though, especially before OCF does its 2021 audit in a couple months.

**Comment by Betree** [comment]
We got [another request](https://opencollective.freshdesk.com/a/tickets/106193) for this on support

**Comment by Betree** [comment]
The [Activity Log](https://github.com/opencollective/opencollective/issues/5195) is partially going to solve this issue: with https://github.com/opencollective/opencollective/issues/5808 we're going to associate the host account with the activity, which means it will continue to appear in the host's activity log even if the collective changes host.

**Comment by Betree** [comment]
Hosts now have access to previously hosted collective's transactions through Dashboard > Ledger, though we still need to ship some improvements for it, like https://github.com/opencollective/opencollective/issues/8218.
